### DISCUSSION TITLE
After GitHub Action deployment successfully, API link showing "SADNESS .." error

### DISCUSSION THREAD

**Discussion by Frank1016** [serverpod/serverpod#669] [Q&A]
Hi guys, I have followed all the steps in Serverpod documentation, and was trying to copy pixorama.live project with same AWS deployment. After successfully deployment through GitHub Actions, I go to "api.techshook.com" and can only see this: 

"SADNESS 2023-02-16 02:00:20.194364Z
serverpod_database: 0.0"

What could be wrong?  Any help will be appreciated!

**Comment by yahu1031** [comment]
@Frank1016 The issue is with the database. Connect to the production DB and create the tables. In case you have done that, Then need to check what the issue is. Use the insights app.

**Reply by Frank1016** [reply]
I have created all the tables in production DB:
<img width="895" alt="Screenshot 2023-02-16 at 3 55 50 PM" src="https://user-images.githubusercontent.com/32797520/219484985-6dc737a4-079f-4a4c-8e7e-a10c1c97080f.png">

If I go to insights link, I also got this error:
<img width="327" alt="Screenshot 2023-02-16 at 3 54 48 PM" src="https://user-images.githubusercontent.com/32797520/219485119-67255be4-3731-4ce3-b78b-cb65d9b4acb5.png">

**Reply by Frank1016** [reply]
@yahu1031 What else should I check to solve the issue?

**Reply by yahu1031** [reply]
@Frank1016 Can you check with [insights app](https://serverpod.dev/insights/Serverpod-1.0.0.zip), so you will find the issue easily. Dm me if you are still stuck. I will share the discussion result here later.

**Reply by yahu1031** [reply]
Also check the server logs tables for any traces in production db.

**Reply by Frank1016** [reply]
> @Frank1016 Can you check with [insights app](https://serverpod.dev/insights/Serverpod-1.0.0.zip), so you will find the issue easily. Dm me if you are still stuck. I will share the discussion result here later.

I just re-deployed everything after "Terraform Destroy". Now if I go to the API link (api.techshook.com), I see a "502 bad gateway" error. I can use Postico to connect my DB (no issue with this connection), but all tables are empty there including the logs table.  The insights app cannot connect anything, always showing "Failed to connect" error:

<img width="994" alt="Screenshot 2023-02-16 at 9 09 44 PM" src="https://user-images.githubusercontent.com/32797520/219531557-eeee7f48-f5c0-4bf7-8e7c-8550adcec1a5.png">

**Reply by yahu1031** [reply]
Check this once #445 @Frank1016

**Reply by Frank1016** [reply]
> Check this once #445 @Frank1016

I followed the steps to change Dart SDK to 2.17.0 (Flutter 3.0.0), but it won't deploy through GitHub Actions:
<img width="1131" alt="Screenshot 2023-02-17 at 10 08 05 AM" src="https://user-images.githubusercontent.com/32797520/219691323-2c7a2531-b24d-4df8-8b9b-fb240e955b52.png">

Then I changed SDK version exactly the same as https://github.com/serverpod/pixorama: Dart SDK to latest in init-script.sh and Flutter SDK to 3.3.2 in deployment-aws.yml,  and then redo the terraform and GitHub actions, now it can be deployed successfully, but all endpoints are still showing "502 bad gateway" error!

I also tried Dart SDK to latest in init-script.sh and Flutter SDK to latest (3.7.3) in deployment-aws.yml, still the same "502 bad gateway" error!

I'm stuck! Please, any help will be greatly appreciated!

**Reply by Frank1016** [reply]
I recreated a new project (the official ServerPod sample with no change), and deployed only the server, and all endpoints are working fine. However, I also want to deploy the website to AWS using Terraform and Github Actions, just like pixorama.live project. Do you know any other sample project like Pixorama that I can look into?

**Reply by yahu1031** [reply]
Not flutter sdk, check the dart sdk and don't use the same version they mentioned, instead you check with the versions you have.
